airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -77,22 +81,92 @@ def get_cluster_id_by_name(self, emr_cluster_name: str, cluster_states: list[str
 
     def create_job_flow(self, job_flow_overrides: dict[str, Any]) -> dict[str, Any]:
         """
-        Creates a job flow using the config from the EMR connection.
-        Keys of the json extra hash may have the arguments of the boto3
-        run_job_flow method.
-        Overrides for this config may be passed as the job_flow_overrides.
+        Create and start running a new cluster (job flow).
+
+        This method use ``EmrHook.emr_conn_id`` for receive initial Amazon EMR cluster configuration.
+        If ``EmrHook.emr_conn_id`` is empty or connection not exists than empty initial configuration is used.
+
+        :param job_flow_overrides: Uses for overwrite parameters in initial Amazon EMR configuration cluster.
+            The resulting configuration will be used in the boto3 emr client run_job_flow method.
+
+        .. seealso::
+            - :ref:`Amazon Elastic MapReduce Connection <howto/connection:emr>`
+            - `API RunJobFlow <https://docs.aws.amazon.com/emr/latest/APIReference/API_RunJobFlow.html>`_
+            - `boto3 emr client run_job_flow method <https://boto3.amazonaws.com/v1/documentation/\
+               api/latest/reference/services/emr.html#EMR.Client.run_job_flow>`_.
         """
-        try:
-            emr_conn = self.get_connection(self.emr_conn_id)
-            config = emr_conn.extra_dejson.copy()
-        except AirflowNotFoundException:
-            config = {}
+        config = {}
+        if self.emr_conn_id:
+            try:
+                emr_conn = self.get_connection(self.emr_conn_id)
+            except AirflowNotFoundException:
+                warnings.warn(
+                    f"Unable to find Amazon Elastic MapReduce Connection ID {self.emr_conn_id!r}, "
+                    "use empty initial configuration. If you want to get rid of this warning "
+                    "message please set `emr_conn_id` to None.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+            else:
+                if emr_conn.conn_type and emr_conn.conn_type != "emr":
```

Comment:
First of all why we need use some magic: `conn_type` not a mandatory attribute for class `airflow.models.connection.Connection` and could be an empty string/None.

Actually it not easy to achieve this, for connection in the UI I thought it almost not possible,
URI base should miss schema something like this `://AKIAIOSFODNN7EXAMPLE:wJalrXUtnFEMI%2FK7MDENG%2FbPxRfiCYEXAMPLEKEY@` or for JSON it might be
```json
{
   "login": "AKIAIOSFODNN7EXAMPLE",
   "password": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
}
```

Connection object do not provide any abilities for actual validate conn_type this. So we could do
1. `if emr_conn.conn_type and emr_conn.conn_type != "emr":`
2. `if (emr_conn.conn_type or "emr") != "emr":`
3. `if emr_conn.conn_type not in ("emr", None):`

Another option. Right now in the PR I suggest that if user not provide any of conn_type, that user mean `emr` however we could do differently if it not set (or empty) than assume that might something went wrong: `if emr_conn.conn_type != "emr":`

Reply:
> the "emr", is there not a Enum, Constant, etc somewhere than can be used instead of this string?

Oh... got you. We have a better than Enum and Constraint -`EmrHook.conn_type` class attribute.

https://github.com/apache/airflow/blob/c94f978a66a7cfc31b6d461bbcbfd0f2ddb2962e/airflow/providers/amazon/aws/hooks/emr.py#L43-L46

Reply:
Change from `"emr"` to `self. conn_type`